### PR TITLE
Switch internal port numbers and route Wi-Fi connect through ExpressJS

### PR DESCRIPTION
**Pull request by maggie0002**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Switched from port 3000 to port 80. It was already port 80 outwards facing, so hasn't made much difference. 

**Which issue (if any) does this pull request address?**
When accessing the device from a URL behind a reverse proxy (i.e. the Balena public URL), it generated an error because it was reaching from one port to another, and seeing it as a security issue. Now that it is defaulted to port 80 this issue no longer exists. 

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by nucleardreamer** [comment]
Ah you beat me to it! I was just looking at doing this as well. The deploy-to-balena workflow is merged now, so if you update you can pin the device on the dashboard to this PR and test it out live

**Comment by nucleardreamer** [comment]
confusing that this looks like it failed, the deploy-to-balena action finished once, but got stuck the second time.

**Comment by maggie0002** [comment]
> confusing that this looks like it failed, the deploy-to-balena action finished once, but got stuck the second time. This could have been just a timeout issue? it looks like the release happened on the balena dashboard

Probably because I am manually pushing to the device to test some things for this commit. Oops, sorry.

**Comment by maggie0002** [comment]
I ended up routing it through the expressJS like you mentioned in the issue ticket. Now all the different services work remotely on the development device you set up via the public URL.

Wi-Fi connect now shouldn't be reachable directly on port 9090 anymore from your browser or Postman, only via the express api. Of course I can't test that one, it will need to be done locally. Similarly, I can't test connecting to a network and disconnecting because if it's using Wi-Fi to allow me access I would get cut off. 

Resolves: https://github.com/maggie0002/balena-device-ui/issues/8

**Comment by nucleardreamer** [comment]
Got it! makes perfect sense. Let me give this a test out tomorrow and see if anything is funky. It looks good in the PR though.

**Comment by maggie0002** [comment]
I also added an internet connectivity endpoint, because I thought I needed it for Wi-FI connect, but turns out I didn't. Left it there anyway for now. Wi-Fi connect on the UI end actually connects to the Wi-Fi connect container to get connectivity status, and instead of checking for internet connectivity it checks to see if a Wi-Fi interface is up. This is better than checking for internet as it allows for LAN connections to be active, and still provides a Wi-Fi connect option.

File: expressjs/src/routes/WifiConnect.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import axios from 'axios'
+import express from 'express'
+import dns = require('dns')
+
+const router = express.Router()
+
+// Set Axios defaults
+const wifiAxios = axios.create({ timeout: 15000 })
+wifiAxios.defaults.baseURL = 'http://py-wifi-connect:9090/'
```
Comment (nucleardreamer) [inline expressjs/src/routes/WifiConnect.ts:9]: removing host networking and using internal service DNS resolved the issue for connecting to py-wifi-connect container, works fine now

**Review by nucleardreamer — APPROVED** [review]

**Review by nucleardreamer — APPROVED** [review]

**Comment by maggie0002** [comment]
Wi-Fi connect runs mounted to the host as it can create a bunch of issues when running on a bridge network. The same problem is seen with the official Balena wifi-connect app, it needs to be mounted to host for best support and compatibility: https://forums.balena.io/t/does-wifi-connect-need-to-be-mounted-to-host-instead-of-port/219349/6?u=maggie

@nucleardreamer were there issues with it using the 172. IP address?

**Comment by nucleardreamer** [comment]
Ah I see, I didn't realize this was the case for the wifi. Sorry about that, I was a little trigger happy! I changed to the service name because the bridge network addresses were changing dynamically (it was on 172.18.0.1, then changed again to 172.23.0.1). Binding to `0.0.0.0` or the loopback didn't work so it seems? I might have missed something.

**Comment by maggie0002** [comment]
> 

Hehe, easily done. Wifi-Connect and I have a long and grim history of problems. It's a tricky needle to thread. 

Bridge networks will change IP address, but when mounted to host the 172.17.0.1 is a default on Linux systems for the host.

**Comment by nucleardreamer** [comment]
I had made that commit because it wasn't working on the test machine with 172, and I couldn't get it working any other way I had tried. How should I fix this up?

**Comment by maggie0002** [comment]
> I had made that commit because it wasn't working on the test machine with 172, and I couldn't get it working any other way I had tried. How should I fix this up?

Which test machine? In what way wasn't it working? Errors in the console, not connecting?

**Comment by maggie0002** [comment]
> Ah I see, I didn't realize this was the case for the wifi. Sorry about that, I was a little trigger happy! I changed to the service name because the bridge network addresses were changing dynamically (it was on 172.18.0.1, then changed again to 172.23.0.1). Binding to `0.0.0.0` or the loopback didn't work so it seems? I might have missed something.

https://forums.balena.io/t/default-subnet-for-intel-nuc-is-missing/352708

**Pull request merged**